react/components/SKUSelector/components/SKUSelector.tsx

```diff
@@ -116,126 +117,195 @@ interface AvailableVariationParams {
   onSelectItemMemo: (callbackItem: CallbackItem) => () => void
   skuItems: SelectorProductItem[]
   hideImpossibleCombinations: boolean
+  possibleToClickOnImpossibleCombination: string[]
   disableUnavailableSelectOptions: boolean
 }
 
-const parseOptionNameToDisplayOption = ({
-  selectedVariations,
-  variationName,
-  skuItems,
-  onSelectItemMemo,
-  imagesMap,
-  variationCount,
-  hideImpossibleCombinations,
-  disableUnavailableSelectOptions,
-}: {
-  selectedVariations: SelectedVariations
-  variationName: string
-  skuItems: SelectorProductItem[]
-  onSelectItemMemo: (callbackItem: CallbackItem) => () => void
-  imagesMap: ImageMap
-  variationCount: number
-  hideImpossibleCombinations: boolean
-  disableUnavailableSelectOptions: boolean
-}) => (variationValue: {
-  name: string
-  originalName: string
-}): DisplayOption | null => {
-  const isSelected = selectedVariations[variationName] === variationValue.name
-  const image = imagesMap?.[variationName]?.[variationValue.name]
+const parseOptionNameToDisplayOption =
```

Comment:
This method seems to be too huge. Can we split it into some meaningful functions?